### DISCUSSION TITLE
troubles connecting a washing machine

### DISCUSSION THREAD

**Discussion by bentouspam** [ofalvai/home-assistant-candy#124] [Device support improvements]
Washing machine Rosières model OBWS69TWMCE-47 bought in August 2021 in France.

Impossible to enter the machine in the Candy integration module.

http://192.168.92.200/http-read.json?encrypted=0
{"response":"BAD REQUEST"}

http://192.168.92.200/http-read.json?encrypted=1
186F6367471012071514152B0F120C1C110B0A0B47591D6B6B686F45390D2B013016081A1010445C4351444B636E646141271B1C475944545454444B636E6461412F080D0D2E02445B435445426967616A40391C47594454434D6B6D676D4F381132014C5F4153444D6C6C6E67463E2406140C02475944545454444B636E646141360C0315415C4451434A6A646D644A301200003613445C4351444B636E6461412D191A54415C4451434A6A646D644A2C121D5C47594456434D6B6D676D4F2713165A4C5F4156444D6C6C6E6746221817564B544753444A6C6B6F6E4C2B1D1C5640534C55414A6B6B686F452114195E41584B5E474F6B6C686844281E105A4A5940594C496E6C6F684329171A5C4F5241524B4268696F6F433212020F094F5241524B4268696F6F4325141E3A46574A534045636F6A6F4425040A310F084F5241505C5B474F6B6C686844350B0939010E074B54475256444D6C6C6E67463F0D000B190B2C07445C4351444B636E64614121010B06083316321507130B46574A534064646C1E6B6C1

$ ./simplyfi 192.168.92.200 getkey
## Candy Simply-Fi tool by Melvin Groenendaal ##
Found key: cbinecffaafgndmh

```
$ ./simplyfi 192.168.92.200 cbinecffaafgndmh read
{
        "statusLavatrice":{
                "WiFiStatus":"0",
                "Err":"255",
                "MachMd":"2",
                "Pr":"2",
                "PrPh":"5",
                "SLevel":"255",
                "Temp":"0",
                "SpinSp":"0",
                "Opt1":"0",
                "Opt2":"0",
                "Opt3":"0",
                "Opt4":"0",
                "Opt5":"0",
                "Opt6":"0",
                "Opt7":"0",
                "Opt8":"0",
                "Steam":"0",
                "DryT":"0",
                "DelVal":"255",
                "RemTime":"10",
                "RecipeId":"0",
                "CheckUpState":"0"
        }
}
```
-----------------------------------------------------------------------------------
integration in HA:
-select Candy integration
-add the ip address
then Error:

```
Candy
Failed to detect encryption, check logs
ask again for the Device IP Address.
```

here are the logs:
```
Logger: custom_components.candy.config_flow
Source: custom_components/candy/client/__init__.py:69
Integration: Candy ([documentation](https://github.com/ofalvai/home-assistant-candy), [issues](https://github.com/ofalvai/home-assistant-candy/issues))
First occurred: 12:04:22 (2 occurrences)
Last logged: 12:06:36

Server disconnected
Traceback (most recent call last):
  File "/config/custom_components/candy/client/__init__.py", line 62, in detect_encryption
    assert resp_json.get("response") != "BAD REQUEST"
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/candy/config_flow.py", line 45, in async_step_user
    encryption_type, key = await detect_encryption(
  File "/config/custom_components/candy/client/__init__.py", line 69, in detect_encryption
    async with session.get(url) as resp:
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client.py", line 559, in _request
    await resp.start(conn)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 898, in start
    message, payload = await protocol.read()  # type: ignore[union-attr]
  File "/usr/local/lib/python3.9/site-packages/aiohttp/streams.py", line 616, in read
    await self._waiter
aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected
```
adding 
logger:
  default: warning
  logs:
    custom_components.candy: debug
`
Logger: custom_components.candy.config_flow
Source: custom_components/candy/client/__init__.py:69
Integration: Candy ([documentation](https://github.com/ofalvai/home-assistant-candy), [issues](https://github.com/ofalvai/home-assistant-candy/issues))
First occurred: 12:32:29 (1 occurrences)
Last logged: 12:32:29

Cannot connect to host 192.168.92.200:80 ssl:default [Connect call failed ('192.168.92.200', 80)]
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/connector.py", line 986, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1056, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1041, in create_connection
    sock = await self._connect_sock(
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 955, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 502, in sock_connect
    return await fut
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 537, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
OSError: [Errno 113] Connect call failed ('192.168.92.200', 80)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/candy/config_flow.py", line 45, in async_step_user
    encryption_type, key = await detect_encryption(
  File "/config/custom_components/candy/client/__init__.py", line 69, in detect_encryption
    async with session.get(url) as resp:
  File "/usr/local/lib/python3.9/site-packages/aiohttp/client.py", line 535, in _request
    conn = await self._connector.connect(
  File "/usr/local/lib/python3.9/site-packages/aiohttp/connector.py", line 542, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/connector.py", line 907, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/usr/local/lib/python3.9/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/usr/local/lib/python3.9/site-packages/aiohttp/connector.py", line 992, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host 192.168.92.200:80 ssl:default [Connect call failed ('192.168.92.200', 80)]
`
I also have a dish wacher RDIN 4S622PS-47       +/- 1 year older but impossible to get the key

http://192.168.92.214/http-read.json?encrypted=0
{"response":"BAD REQUEST"}
http://192.168.92.214/http-read.json?encrypted=1
186A646C4F1D1000101D122D320C1504415D156867676D43371C001D0A3A0F2A0A4554475D4C486C6E61684B2602020500022B171F0116044652432C554F4A61696E67473E1A05150B2C36081605445641554C4960646D684625041D042E071E0A040147574C544348656B606C4F351802151A36190114435E4A514B49606C656A453A1708070A340A074353475D44406E6D676C4F2B070E4652435947416B666A6E4C351F01031305054353473D5E4E4F6A646C644C2119101A002D1714445641574C4960646D684627110C0B29090311281E114F54465046446C636C644428060B0F1C3E1A0513104A5B4B554F4A61696E67473F0B09350D05044B5F4F545C5645426867676D432901121A360C0A18415D4C554F42696B6D6143240C1E153E0A091D004F54465146446C636C64442313020021020116435E4A504B49606C656A453C001E0B10435E4A514B49606C656A452D0D080D0F34144A5B4B554F4A61696E67471F5F465B4659434568676F6541155C47574C544348656B606C4F145F415D4C564F42696B6D61431B514F5C4E5245426867676D43165D4353475D44406E6D676C4F1C52435E4A504B49606C656A451C524F54465446446C636C64441E5B4554475C4C486C6E61684B1754445641293B29214C486C6E61684B175C534E59455F4760646D1C69621C

$ ./simplyfi 192.168.92.214 getkey
## Candy Simply-Fi tool by Melvin Groenendaal ##
error: find_xor_key_list, could not find key

I don't know if this is possible to make it working... ???

Home Assistant Core 2022.5.5
Home Assistant Supervisor 2022.05.3
Home Assistant OS 8.1
Kernel version 5.15.32-v8
Agent version 1.2.1
HW Raspberry PI 4b

Regards from Paris, France.
Benoit.

**Comment by fire90** [comment]
Hello,

i have the same model and i just add the ip address. here is what i get : 
<img width="456" height="164" alt="image" src="https://github.com/user-attachments/assets/5718089a-dfbd-4f3f-91c0-47e5ad071829" />
